document/CoreFormat.mdk

```diff
@@ -2439,9 +2440,35 @@ C++: `"std::runtime_error"`
 C#: `"System.ArgumentNullException"`
 ~
 
+If the tool does not have access to information about the object that was thrown, the `type` property shall be absent.
+
+~ IsoNote
+This will be the case in C++ programs that throw objects that do not derive from `std::exception`:
```

Comment:
Hmm... I'm a bit divorced from the context of the format at this time to make a statement like that. Is this supposed to mean something like "an exception occurred inside the analyzer" ? Or is it supposed to be something like "I was analyzing some user code, and the user code threw an exception" ?

If the former, then I wouldn't say much about how to fill those in -- C++ analyzers would probably just have one exception type like "INTERNAL COMPILER ERROR C1001" or whatever, and the message would probably be the same every time. If you get a log at all, which is unlikely.

If the latter, then I'm not sure. If it's a debugger or something that data is going to be known. I would just say tools should make a best-effort attempt at filling those in, which will be trivially easy in languages with strong code metadata. (A Java programmer doesn't need a crazy explanation of what an exception is. A C++ programmer won't be confused by not having type information there, etc.)

As for signals, you could call them "exceptions" and say the type is the signal name (translating a signal to a name is easy because they're all documented in signal.h) or similar. Just note that many signals are not catchable so you're not going to see them if they occur in the analyzer....

I've not ever used signals myself though so I would try to get opinions from folks who have before saying anything about them.